### ISSUE TITLE
Time so lang 

### ISSUE THREAD

**Issue by y912765390** [antlr/antlr4#2861]
var antlr4 = require('antlr4');

var MyGrammarLexer = require('./Java8Lexer').Java8Lexer;
var MyGrammarParser = require('./Java8Parser').Java8Parser;
var MyGrammarListener = require('./Java8ParserListener').Java8ParserListener;
var chars = new antlr4.InputStream(input);
var lexer = new MyGrammarLexer(chars);
var tokens = new antlr4.CommonTokenStream(lexer);
var parser = new MyGrammarParser(tokens);
parser.buildParseTrees = true;
console.time('aa');
var tree = parser.compilationUnit();
console.timeEnd('aa');
When I used java8.g4， parser.compilationUnit();The execution time is too long. A 70-80 line java code will take 50 seconds or more to execute.
I don't know if I wrote something wrong or if it was slow to build the tree. Can you give me some advice

**Comment by KvanTTT** [comment]
Try to use [java](https://github.com/antlr/grammars-v4/tree/master/java/java) grammar - it's much faster and also supports Java 8.

**Comment by y912765390** [comment]
I know that java files can be generated, and the build tree is faster. I tried, but this is a front-end project, and the back-end is done with node. So I don't want to use the Java environment. At the same time, I also hope that the time can be reduced to 1-2 seconds. Can it be done

**Comment by KvanTTT** [comment]
I mean trying another java grammar, not another runtime.

**Comment by y912765390** [comment]
I'm sorry, I don't quite understand. Do you mean to use another Java.g4 file？
I just want to be able to quickly generate this syntax tree in the JS environment. If it is what I understand, change a G4 file. Can you send it to me?
&nbsp;Thank you very much for your help. I try to parse the G4 generated java file and generate JS very slowly, which makes me puzzled

**Comment by KvanTTT** [comment]
> I'm sorry, I don't quite understand. Do you mean to use another Java.g4 file？

Yes, I wrote about the link in the comment above [java](https://github.com/antlr/grammars-v4/tree/master/java/java).

**Comment by y912765390** [comment]
Thank  you very much.------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;&quot;Ivan&amp;nbsp;Kochurkin&quot;<notifications@github.com&gt;
发送时间:&nbsp;2020年7月10日(星期五) 下午5:56
收件人:&nbsp;&quot;antlr/antlr4&quot;<antlr4@noreply.github.com&gt;;
抄送:&nbsp;&quot;y912765390&quot;<912765390@qq.com&gt;;&quot;Author&quot;<author@noreply.github.com&gt;;
主题:&nbsp;Re: [antlr/antlr4] Time so lang  (#2861)
